tutorials/foreign-object.html

```diff
@@ -0,0 +1,813 @@
+            <a href="https://developer.mozilla.org/en-US/docs/Web/SVG/Element/style" target="_blank"><code>style</code></a> element, or inline 
+            styles via the <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/style" target="_blank">style</a> attribute 
+            if you prefer, but as we are adding a substantial amount of CSS, we will use a separate stylesheet.
+        </p>
+
+        <pre><code>.outer {
+    background: linear-gradient(92.05deg, hsl(355, 100%, 68%) 12.09%, hsl(355, 100%, 63%) 42.58%, hsl(355, 85%, 54%) 84.96%);
+    padding: 6px;
+    width: 100%;
+    height: 100%;
+}
+
+.inner {
+    background-color: hsl(210deg 36.67% 11.76%);;
+    width: 100%;
+    height: 100%;
+}
+
+.form {
+    display: flex;
+    font-family: sans-serif;
+    flex-direction: column;
+    justify-content: space-evenly;
+    padding: 0 32px;
+    width: 100%;
+    height: 100%;
+}
+
+input {
+    all: unset;
+    background: hsl(218, 80%, 2%);
+    color: white;
+    cursor: text;
+    font-size: 1rem;
+    height: 56px; 
+    outline-color: hsl(355, 100%, 63%);
+    outline-width: thin;
+    padding: 0 32px;
+}
+
+input::placeholder {
+    color: hsl(0, 0%, 50%);
+}
+
+input:focus-visible {
+    outline-style: inset;
+}
+
+button {
+    background-color: hsl(218, 80%, 2%);
+    border: unset;
+    cursor: pointer;
+    font-family: sans-serif;
+    font-size: 1rem;
+    font-weight: 600;
+    height: 56px;
+    outline-color: hsl(355, 100%, 63%);
+    outline-width: thin;
+}
+
+button:focus-visible {
+    outline-style: inset;
+}
+
+button span {
+    background: linear-gradient(92.05deg, hsl(355, 100%, 68%) 12.09%, hsl(355, 100%, 63%) 42.58%, hsl(355, 85%, 54%) 84.96%);
+    -webkit-background-clip: text;
+    background-clip: text; 
+    -webkit-text-fill-color: hsl(0deg 0% 0% / 0%);
+    font-size: 1.2rem;
+}
+</code></pre>
+
+        <div class="paper" id="paper-foreign-object-prevent-default"></div>
+        <p>JointJS source code: <a href="js/foreign-object.js" target="_blank">foreign-object.js</a></p>
+
+
+        <p>
+            The joy of working with <code>foreignObject</code> in our shapes is that we still get to take advantage of native web functionality.
+            When using a <code>form</code>, you might be used to programmatically associating a <code>label</code> with an <code>input</code> 
+            by providing an <code>'id'</code>
+            attribute to the <code>input</code>, and a <code>'for'</code> attribute to the <code>label</code> which has the same value. This has
+            a number of advantages such as screen readers reading the label when the associated input is focused, and also that the focus is 
+            passed to the input when the associated label is clicked.
+        </p>
+
+        <p>
+            On the other hand, when working with JointJS, you will have to make decisions about how some HTML elements lose focus. If a user clicks 
```

Comment:
"~~On the other hand,~~ When working with JointJS, you still need to make some decisions..."